### PR TITLE
Enhancement/parking debugger player position

### PR DESCRIPTION
**Pull request by Gursaanj**
parking spot Debugger

- Allows you to auto select the parking spot closest to selected player
- Looks prettier (more presentable haha)
- Made a GUISkin library we can use for any new/other editor windows we might want

### CODE REVIEW

File: Assets/Editor/PartyParkadeParkingDebugger.cs [hunk 1]
```diff
@@ -7,25 +7,36 @@
...
+    private Rect m_titleRect;
     private Rect m_scrollViewRect;
-    private Vector2 m_scrollView = new Vector2(1, 1);
-    private int m_gridXCount = 3;
+    private Rect m_playerSelectionRect;
     private Rect m_lowerRect;
-    private float m_buttonOffset = 30f;
+    #endregion
+
+    private Vector2 m_scrollView = new Vector2(1, 1);
```
Comment (m4rtint) [inline Assets/Editor/PartyParkadeParkingDebugger.cs:25]: Vector2.One?

**Comment by Gursaanj** [comment]
[(2) Refactor parking debug tool based on comments from the meeting](https://trello.com/c/59IraHOy/525-2-refactor-parking-debug-tool-based-on-comments-from-the-meeting)

File: Assets/Editor/PartyParkadeParkingDebugger.cs [hunk 2]
```diff
@@ -79,44 +94,104 @@ private void InitializeLevelData()
...
+
     private void CreateSelectionGrid()
     {
-        m_scrollView = GUILayout.BeginScrollView(m_scrollView, GUILayout.Width(m_scrollViewRect.width), GUILayout.Height(m_scrollViewRect.height));
+        GUILayout.BeginArea(m_scrollViewRect);
+
         if (m_parkingSpots != null && m_parkingSpotNames != null && m_currentParkingSpotIndex > -1 && m_currentParkingSpot != null && m_parkingTimer != null)
         {
+            GUILayout.BeginHorizontal();
+            GUILayout.Label("Select a spot from the grid below while", skin.GetStyle("SubHeader"), GUILayout.Height(m_subHeaderHeight));
```
Comment (m4rtint) [inline Assets/Editor/PartyParkadeParkingDebugger.cs:119]: "SubHeader" placed ontop?

File: Assets/Editor/PartyParkadeParkingDebugger.cs [hunk 3]
```diff
@@ -79,44 +94,104 @@ private void InitializeLevelData()
...
                 m_parkingTimer.DebugParkingSpotChanged(m_currentParkingSpot);
                 m_parkingTimer = m_currentParkingSpot.GetComponent<ParkingTimer>();
             }
         }
         else
         {
-            if (GUILayout.Button("No ParkingSpots Detected, Close the window??", GUILayout.Height(m_lowerRect.height)))
+            if (GUILayout.Button("No Parking Spots Detected, Close the window??", skin.GetStyle("button"), GUILayout.Height(m_lowerRect.height)))
             {
                 Debug.Log("Parking Debugger Window has been closed");
```
Comment (m4rtint) [inline Assets/Editor/PartyParkadeParkingDebugger.cs:196]: Better to remove this Debug.Log?

File: Assets/Editor/PartyParkadeParkingDebugger.cs (hunk 2 above)
Comment (Gursaanj) [reply Assets/Editor/PartyParkadeParkingDebugger.cs:119]: do you mean declare a string for "SubHeader"?

Comment (m4rtint) [reply Assets/Editor/PartyParkadeParkingDebugger.cs:119]: Yeah, but I'm not opposed to leaving it as is, up to you

Comment (Gursaanj) [reply Assets/Editor/PartyParkadeParkingDebugger.cs:119]: might as well!!

**Pull request merged**